Code/Engine/GameEngine/Gameplay/SceneTransitionComponent.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <Core/World/EventMessageHandlerComponent.h>
+#include <GameEngine/GameEngineDLL.h>
+
+struct ezMsgTriggerTriggered;
+
+using ezSceneTransitionComponentManager = ezComponentManager<class ezSceneTransitionComponent, ezBlockStorageType::Compact>;
+
+/// \brief What ezSceneTransitionComponent should do when it gets triggered through a ezMsgTriggerTriggered
```

Comment:
Would it make more sense to have one scene transition component that can handle both the preload and scene switch. The trigger messages can have a string to differentiate what each trigger should do. The expected trigger messages could be exposed as properties on this component with a default value so they can be customized but don't have to.
I think that would make it easier to setup and keep all the logic in one place.

Reply:
Hm, maybe. I mean you can already do that now, if you add a visual script. I think the current setup is the easiest to use and understand. Not sure yet, it would be easy to change, though.